buildscripts/builddrivers.py

```diff
@@ -0,0 +1,291 @@
+#!/usr/bin/python
+#########################################################################################
+#
+# Description:  This script helps to build drivers in a Windows environment for PHP 7+ (32-bit/64-bit)
+#
+# Requirement:
+#               python 3.4
+#               PHP SDK and PHP Source 
+#               Driver source code folder / GitHub repository
+#               Visual Studio 2015 (PHP 7.0* and 7.1*) and Visual Studio 2017 (PHP 7.2*)
+#
+# Execution: Run with command line with required options.
+# Examples: 
+#           py builddrivers.py (for interactive mode)
+#           py builddrivers.py -v=7.1.7 -a=x86 -t=ts -d=all -g=no
+#           py builddrivers.py --PHPVER=7.0.22 --ARCH=x64 --THREAD=nts --DRIVER=all --GITHUB=yes
+#
+# Output: Build the drivers using PHP SDK. When running locally, if build is unsuccessful, 
+#         the log file will be launched for examination. Otherwise, the drivers will be renamed 
+#         and copied to the designated location(s).
+#
+#############################################################################################
+
+import sys
+import shutil
+import os.path
+import argparse
+from buildtools import BuildUtil
+
+class BuildDriver(object):
+    """Build sqlsrv and/or pdo_sqlsrv drivers with PHP source with the following properties:
+    
+    Attributes:
+        util            # BuildUtil object whose constructor takes phpver, driver, arch, thread, debug 
+        repo            # GitHub repository
+        branch          # GitHub repository branch
+        download_source # download source from GitHub or not
```

Comment:
all these options seem so intertwined they overlap and have conflicting meanings. Eg. specifying github/branch and download_source = no and/or local/source_path makes no sense at all these are mutually exclusive options. The flags should be internal only they do not need to be externally visible. I think you only need 3 options (actually 2 if you combine gthub/branch) to convey this information. If Github/branch then download_source is obviously yes there's no need to specify that.  If source_path then obviously it's local no need to ask whether build is local or not. Actually, there's no need for the concept or "local" or "remote" at all. Basically there are 2 ways you can get the source, you either specify the path to the source (which could be local or remote unc path), or you specify the github repo.